common-custom-user-data-maven-extension/src/main/java/com/gradle/ApiAccessor.java

```diff
@@ -43,19 +39,18 @@ private static void ensureBuildScanApiIsAccessible(Class<?> extensionClass) thro
         }
     }
 
-    private static BuildScanApi lookupBuildScanApi(PlexusContainer container) throws MavenExecutionException {
-        if (!container.hasComponent(BUILD_SCAN_API_CONTAINER_OBJECT)) {
+    private static <T> T lookupClass(Class<T> componentClass, PlexusContainer container) throws MavenExecutionException {
+        if (!container.hasComponent(componentClass)) {
             return null;
         } else {
             try {
-                return (BuildScanApi) container.lookup(BUILD_SCAN_API_CONTAINER_OBJECT);
+                return componentClass.cast(container.lookup(componentClass));
```

Comment:
This should be enough, right?

```suggestion
                return container.lookup(componentClass);
```

Reply:
Done

Reply:
Done.